src/ffi.rs

```diff
@@ -778,6 +771,37 @@ fn check_geos_predicate(val: i32, p: PredicateType) -> GeosResult<bool> {
     }
 }
 
+fn check_same_geometry_type(geoms: &[GGeom], geom_type: GEOSGeomTypes) -> bool {
+    geoms.iter()
+        .map(|g|
```

Comment:
I think you can rewrite this like:

```rust
geoms.iter().all(|g| g.geometry_type == Ok(geom_type))
```

Reply:
Ah great :+1:  I was looking for some short-circuiting method like `all` (so I go with `collect` and it ended up a bit messy)

Reply:
I somehow changed it but it can't be as clean as in your example because we still have to verify that `geometry_type` does not return an error.

Reply:
oh you're right, it seems that the `Error` is not comparable sorry :roll_eyes:  

It might be a bit more rustic to write this:
```rust
    geoms
        .iter()
        .all(|g| g.geometry_type().map(|t| t == geom_type).unwrap_or(false))
```

Reply:
Respect :clap:  I struggled with the various `unwrap_*` and `map_*` for a while and I couldn't find a clean solution!